### PR TITLE
Audio

### PR DESCRIPTION
**Pull request by ScottyXIII**
The audio is started in the scene Create() function but it only seems to fire when you click on the browser... Feels like it's doing it onFocus event.

### CODE REVIEW

File: src/scenes/game-scene.ts [hunk 1]
```diff
@@ -46,29 +48,61 @@ const levelConfig = {
   ],
 };
 
+const soundConfig = [
+  {
+    key: 'punch',
+    filePath: 'audio/sfx/punch.wav',
```
Comment (norgeous) [inline src/scenes/game-scene.ts:54]: im not 100% sure, but i think these file paths have to start with `./` or they dont work after deployment

File: src/objects/Audio.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  static preload(scene: Phaser.Scene, configs: AudioConfigType[]) {
+    for (let i = 0; i < configs.length; i += 1) {
+      const config: AudioConfigType = configs[i];
+      scene.load.audio(config.key, config.filePath);
+    };
+  }
+
+  playAudio(key: string) {
+    if (this.audio[key] !== undefined && this.audioConfig[key] !== undefined)
```
Comment (norgeous) [inline src/objects/Audio.ts:32]: how would you feel about:
```
if (!!this.audio[key] && !!this.audioConfig[key])
```
it's equivalent (for this case)

File: src/scenes/game-scene.ts [hunk 3]
```diff
@@ -46,29 +48,61 @@ const levelConfig = {
...
+      this.audio?.playAudio("punch");
+    });
 
     this.matter.add.mouseSpring();
 
     this.parallax = new Parallax(this, parallaxName);
     this.level = new Level(this, levelConfig);
+    this.audio = new Audio(this, soundConfig);
+
+    this.audio?.playAudio("music2");
```
Comment (norgeous) [inline src/scenes/game-scene.ts:105]: its not linted :thinking: 
please run `npm run lint` and commit the changes (mapData1.json is also not linted, my bad)
also i will help with that setup on your local next couple of days

File: src/scenes/game-scene.ts [hunk 4]
```diff
@@ -46,29 +48,61 @@ const levelConfig = {
...
 
+const soundConfig = [
+  {
+    key: 'punch',
+    filePath: 'audio/sfx/punch.wav',
+    loop: false
+  },
+  {
+    key: 'music1',
+    filePath: 'audio/music/Fluffing-a-duck.mp3',
```
Comment (norgeous) [inline src/scenes/game-scene.ts:59]: might be a good idea to record the source for the mp3s in the README?

i should also add the source of the parallaxes (that is "parallax" search on itch.io)

**Review by norgeous — APPROVED** [review]

File: src/objects/Audio.ts (hunk 2 above)
Comment (ScottyXIII) [reply src/objects/Audio.ts:32]: ah yes I would prefer !! as I use it in PHP. It's less text to read! Didn't think I could do it with JS lol

**Pull request merged**